src/index.ts

```diff
@@ -210,7 +217,12 @@ interface PoolEvents {
   error: (error: Error, client?: PoolClient) => void;
 }
 
+interface DebugEvents {
+  receiveConnectionDelay: (startTime: [number, number], endTime: [number, number]) => void;
+}
+
 type PoolEmitter = StrictEventEmitter<EventEmitter, PoolEvents>;
+type DebugEmitter = StrictEventEmitter<EventEmitter, DebugEvents>;
```

Comment:
With v5, I added parameters to the `connectionAddedToPool` event. The `startTime` parameter for that event should allow you to determine how long it took to establish a connection to the server.